### PR TITLE
Tosa folders for bitwise_and, bitwise_or, greater_equal, equal, log, maximum, minimum

### PR DESCRIPTION
**Pull request by ttjost**

### CODE REVIEW

File: mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp [hunk 1]
```diff
@@ -48,8 +48,7 @@ OffsetType indexToOffset(DimensionType shape, DimensionType index) {
   return offset;
 }
 
-SmallVector<int64_t> offsetToIndex(DimensionType shape,
-                                               OffsetType offset) {
+SmallVector<int64_t> offsetToIndex(DimensionType shape, OffsetType offset) {
```
Comment (ttjost) [inline mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp:51]: I am fixing this! No need to reformat.

File: mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp [hunk 2]
```diff
@@ -1090,6 +1090,174 @@ struct TosaFoldConstantErf
...
+  }
+
+  DenseElementsAttr computeFloat(DenseElementsAttr lhsValues,
+                                 DenseElementsAttr rhsValues,
+                                 PatternRewriter &rewriter,
+                                 MinimumOp op) const {
+    return applyElementWise<APFloat, APFloat>(
+        lhsValues, rhsValues, op.getType(),
+        [](const APFloat &first, const APFloat &second) {
+          return first < second ? first : second;
```
Comment (mgehre-amd) [inline mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp:1228]: Did you check that this definition of minimum has the same behavior with respect to inf and nan as specified by tosa?

**Review by mgehre-amd — APPROVED** [review]

File: mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp (hunk 2 above)
Comment (ttjost) [reply mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp:1228]: Very good point. I'll check it.

Comment (ttjost) [reply mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp:1234]: FYI, you're right. We were not handling NaNs properly. Now it should be fixed in `greater`, `greater_equal`, `minimum`, and `maximum`.

File: mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp [hunk 3]
```diff
@@ -1090,6 +1090,174 @@ struct TosaFoldConstantErf
...
+  }
+
+  DenseElementsAttr computeFloat(DenseElementsAttr lhsValues,
+                                 DenseElementsAttr rhsValues,
+                                 PatternRewriter &rewriter,
+                                 MaximumOp op) const {
+    return applyElementWise<APFloat, APFloat>(
+        lhsValues, rhsValues, op.getType(),
+        [](const APFloat &first, const APFloat &second) {
+          return first > second ? first : second;
```
[the remark this answers is not in the document]
Comment (ttjost) [reply mlir/lib/Dialect/Tosa/Transforms/TosaFolders.cpp:1264]: resolving since I commented above. =)

**Review by mgehre-amd — APPROVED** [review]

**Pull request merged**